tests/test_dereference.py

```diff
@@ -106,6 +106,9 @@ class Group(Document):
             [m for m in group_obj.members]
             self.assertEqual(q, 2)
 
+            [m for m in group_obj.members]
```

Comment:
its minor but could you add a simple comment above this, something like
`# verifies that no additional queries gets executed if we re-iterate over the ListField`
A newcomer might not get that its intended and think its a copy/paste mistake

Reply:
@bagerard Added comment in https://github.com/MongoEngine/mongoengine/pull/2007/commits/fdcaca42ae6145ac102ec3475bdc8c550a7a604a